### DISCUSSION TITLE
Is it possible to use OutputExpression in OnFailure Action?

### DISCUSSION THREAD

**Discussion by akashskypatel** [microsoft/RulesEngine#331] [Q&A] [answered]
Is it possible to use an OutputExpression for OnFailure condition when using post rule execution actions? When I try to use below example I am getting a failure condition without any result. Simple example below. I know I might be able to achieve the result using a chained rule but I am specifically wondering if we can use OutputExpressions in OnFailure action. Would cut down on having to write separate rules for failure conditions that need expression evaluation.

Rule:
```
[
  {
    "WorkflowName": "inputWorkflow",
    "Rules": [
      {
        "RuleName": "GiveDiscount10Percent",
        "Expression": "input1.Base == \"252\" AND input1.Exemption != \"Exempt\"", // AND input1.Exemption != \"Exempt\"
        "LocalParams": [
          {
            "Name": "fee1",
            "Expression": "0.08"
          },
          {
            "Name": "fee2",
            "Expression": "0.01"
          }
        ],
        "Actions": {
          "OnSuccess": {
            "Name": "OutputExpression", //Name of action you want to call
            "Context": { //This is passed to the action as action context
              "Expression": "input1.Quantity * fee1"
            }
          },
          "OnFailure": {
            "Name": "OutputExpression", //Name of action you want to call
            "Context": { //This is passed to the action as action context
              "Expression": "input1.Quantity * fee2"
            }
          }
        }
      }
    ]
  }
]
```

Input:
```
[
  {
    "Parameter": {
      "Base": "252",
      "Quantity": 248,
      "Exemption": "Exempt"
    },
    "InputRule": "input1"
  }
]
```

**Comment by abbasc52** [comment, marked as answer]
There is no separate implementation for `onfailure`. If it works for `onsuccess` it should work for `onfailure`
If you can share a working sample on https://dotnetfiddle.net/ i could have a look
https://github.com/microsoft/RulesEngine/blob/4a2b345fe9e52ac5c4a856057b7815fd4df990da/src/RulesEngine/RulesEngine.cs#L130-L155

**Reply by akashskypatel** [reply]
Apologies, it was my mistake. Seems like I was having problems because of commented expression in json string. It does work if I remove the comment. Thanks for taking a look.
